### PR TITLE
Item functionalities

### PR DESCRIPTION
**Pull request by buddja**
Fixed item functionality in both combat and world state, as well as issue #145.

### CODE REVIEW

**Comment by bitcraft** [comment]
Thanks for taking on this issue.  Everything looks good except the changes in combat.py.  The test you have for Techniques is going to fail all the time (the "is" operator), so it needs to be removed.  I would revert the changes to compat.py and add an empty method advance_round to the Item class.  Items and techniques should share the same interface, since it simplifies how they are used in the code (sorry if that is not clear or is omitted in the docstrings).  Eventually, multi-round use items can be implemented this way, too.  (I have actually done all these things locally, but you beat be to making a PR for it :)

I like that you added the dialogs after using items in the World State.  If you make those changes, I don't see a problem merging it.

In item.py in the Item class:

```python
def advance_round():
    pass
```

The changes in item.py line 59 can be simplified as follows:

```python
    def determine_state_called_from(self):
        # hackish way to just get this working...  may come back to this later, idk
        return self.game.active_states[1].name
```

**Comment by buddja** [comment]
Made the suggested changes, I believe everything should be good.
If there are any more suggestion's i'll gladly make them until then i'll continue moving down the issue tracker.

File: tuxemon/core/states/combat/combat.py [hunk 1]
```diff
@@ -499,7 +499,9 @@ def perform_action(self, user, technique, target=None):
 
         :returns:
         """
-        technique.advance_round()
+        if technique is Technique:
+            technique.advance_round()
+            
         result = technique.use(user, target)
```
Comment (bitcraft) [inline tuxemon/core/states/combat/combat.py]: Please remove this test.  :)

**Comment by bitcraft** [comment]
Go ahead and revert combat.py and I'll merge it.  Thanks!

**Comment by bitcraft** [comment]
I'm going to apologize here for a sec.  The complex way you were searching for the CombatState in the item screen was actually better than the simple one.  I tested it a few times and prefer your way.  Open a PR if you'd like to have it merged in.  Thanks again!

**Comment by buddja** [comment]
I will put it in my next round of pull requests, probably sometime next week.

**Comment by bitcraft** [comment]
Sounds great!  Just a FYI, I'm doing a bit of item work, so that the
capture device is useable.  The core issue has been establishing a way to
select targets for techniques and items.  #129 #132  I'm generally on
freenode #tuxemon if you'd like to try to coordinate things with chat.

**Pull request merged**